src/backend/gpopt/translate/CTranslatorDXLToPlStmt.cpp

```diff
@@ -4189,6 +4201,19 @@ CTranslatorDXLToPlStmt::TranslateDXLDml(
 							 NULL,	// translate context for the base table
 							 child_contexts, output_context);
 
+	// set targetlist as list of target entries to be computed for parent nodes
+	plan->targetlist =
+		TranslateDXLProjList(project_list_output_dxlnode, &base_table_context,
+							 child_contexts, output_context);
+
+	m_has_returning = m_has_returning || plan->targetlist != NIL;
+
+	if (plan->targetlist != NIL &&
```

Comment:
Is it possible that m_has_returning is true without target list? What about
```suggestion
	m_has_returning = (plan->targetlist != NIL);

	if (m_has_returning &&
```
?

Reply:
It can be possible if we have multiple dml nodes in plan
But currently GPDB only accepts plans with no more than one modifying node so this logic is for future, when GPDB will be able to handle multiple dml nodes

Reply:
done